temporal-sdk/src/main/java/io/temporal/client/WorkflowStubImpl.java

```diff
@@ -131,45 +131,71 @@ public WorkflowExecution start(Object... args) {
   }
 
   @Override
-  public <R> WorkflowUpdateHandle<R> updateWithStart(
-      UpdateWithStartWorkflowOperation<R> updateOperation, Object... startArgs) {
+  public <R> WorkflowUpdateHandle<R> startUpdateWithStart(
+      @Nonnull UpdateOptions<R> updateOptions,
+      Object[] updateArgs,
+      @Nonnull WithStartWorkflowOperation<?> startOp) {
     if (options == null) {
-      throw new IllegalStateException("Required parameter WorkflowOptions is missing");
+      throw new IllegalStateException(
+          "Required parameter WorkflowOptions is missing in WorkflowStub");
     }
-    if (!updateOperation.markInvoked()) {
-      throw new IllegalStateException("UpdateWithStartWorkflowOperation was already executed");
+    if (options.getWorkflowIdConflictPolicy() == null) {
+      throw new IllegalStateException(
+          "Required parameter WorkflowIdConflictPolicy in WorkflowOptions is missing in WorkflowStub");
     }
-    checkExecutionIsNotStarted();
+    if (!startOp.markInvoked()) {
+      throw new IllegalStateException("WithStartWorkflowOperation was already executed");
+    }
+    if (startOp.getStub() == null) {
+      throw new IllegalStateException(
+          "WithStartWorkflowOperation was created with different WorkflowStub");
+    }
+    updateOptions.validate();
+
     String workflowId = getWorkflowIdForStart(options);
     WorkflowExecution workflowExecution = null;
     try {
+      // gather inputs
       WorkflowClientCallsInterceptor.WorkflowStartInput startInput =
           new WorkflowClientCallsInterceptor.WorkflowStartInput(
-              workflowId, workflowType.get(), Header.empty(), startArgs, options);
+              workflowId, workflowType.get(), Header.empty(), startOp.getArgs(), options);
+      WorkflowClientCallsInterceptor.StartUpdateInput<R> updateInput =
+          startUpdateInput(
+              updateOptions,
+              updateArgs,
+              WorkflowExecution.newBuilder().setWorkflowId(workflowId).build());
       WorkflowClientCallsInterceptor.WorkflowUpdateWithStartInput<R> input =
           new WorkflowClientCallsInterceptor.WorkflowUpdateWithStartInput<>(
-              startInput, updateOperation);
+              startInput, updateInput);
+
       WorkflowClientCallsInterceptor.WorkflowUpdateWithStartOutput<R> output =
           workflowClientInvoker.updateWithStart(input);
+
+      // gather outputs
       workflowExecution = output.getWorkflowStartOutput().getWorkflowExecution();
       populateExecutionAfterStart(workflowExecution);
```

Comment:
In other SDKs where we'll have a Future to obtain the workflow handle from the start operation, that future is immediately fulfilled once the run id is known. This can happen before the update wait stage has been reached.

I wonder if we want to do sth similar here. We could assign the workflow execution to the stub as soon as the run id is known. But it's not clear to me that it would do much since calling other functions on the stub are not waiting for the execution to be available.

I think we shouldn't do anything right now and re-consider based on feedback.